pkg/swagger/paths.go

```diff
@@ -105,8 +128,60 @@ func (g *generator) populateResponses(typ string, isDelete bool, statusCodes ...
+			Responses:   g.populateResponses(titleCaser.String(childResourceType)+"List", false, http.StatusOK),
+			Pageable: &Pageable{
+				NextLinkName: "nextLink",
+			},
+		},
+	}
+	ps["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RedHatOpenShift/openshiftclusters/{resourceName}/"+childResourceType+"/{childResourceName}"] = &PathItem{
+		Get: &Operation{
+			Tags:        []string{titleCaser.String(childResourceType) + "s"},
+			Summary:     "Gets a " + friendlyName + " with the specified subscription, resource group and resource name.",
+			Description: "The operation returns properties of a " + friendlyName + ".",
+			OperationID: titleCaser.String(childResourceType) + "s_Get",
+			Parameters:  g.populateParameters(7, titleCaser.String(childResourceType), friendlyName),
+			Responses:   g.populateResponses(titleCaser.String(childResourceType), false, http.StatusOK),
+		},
+		Put: &Operation{
+			Tags:        []string{titleCaser.String(childResourceType) + "s"},
+			Summary:     "Creates or updates a " + friendlyName + " with the specified subscription, resource group and resource name.",
+			Description: "The operation returns properties of a " + friendlyName + ".",
+			OperationID: titleCaser.String(childResourceType) + "s_CreateOrUpdate",
+			Parameters:  g.populateParameters(8, titleCaser.String(childResourceType), friendlyName),
+			Responses:   g.populateResponses(titleCaser.String(childResourceType), false, http.StatusOK, http.StatusCreated),
+		},
+		Delete: &Operation{
+			Tags:        []string{titleCaser.String(childResourceType) + "s"},
+			Summary:     "Deletes a " + friendlyName + " with the specified subscription, resource group and resource name.",
+			Description: "The operation returns nothing.",
+			OperationID: titleCaser.String(childResourceType) + "s_Delete",
+			Parameters:  g.populateParameters(7, titleCaser.String(childResourceType), friendlyName),
+			Responses:   g.populateResponses(titleCaser.String(childResourceType), true, http.StatusOK, http.StatusNoContent),
+		},
+		Patch: &Operation{
+			Tags:        []string{titleCaser.String(childResourceType) + "s"},
+			Summary:     "Patches (create or update) a " + friendlyName + " with the specified subscription, resource group and resource name.",
+			Description: "The operation returns properties of a " + friendlyName + ".",
+			OperationID: titleCaser.String(childResourceType) + "s_Update",
+			Parameters:  g.populateParameters(9, titleCaser.String(childResourceType), friendlyName),
+			Responses:   g.populateResponses(titleCaser.String(childResourceType), false, http.StatusOK, http.StatusCreated),
+		},
+	}
+}
+
 // populateTopLevelPaths populates the paths for a top level ARM resource
-func (g *generator) populateTopLevelPaths(resourceProviderNamespace, resourceType, friendlyName string) (ps Paths) {
+func (g *generator) populateTopLevelPaths(resourceProviderNamespace string, resourceType string, friendlyName string) (ps Paths) {
```

Comment:
nit: was there an issue with the type consolidation, or could this be reverted back?

Reply:
If I didn't add the type would it make sense to you what type those values were?

Reply:
yes, https://go.dev/ref/spec#Function_types

Reply:
`resources` in our code base can refer to structs, azure resources, arm resources, etc... I still think that its easier to read and understand with my change. If you think this makes the function more confusing, less maintainable, and less readable I can revert it, but I stand by it.

Reply:
meh. I'll revert it. Maybe the next guy will just be smarter than me :)

Reply:
I don't think it makes it any more or less confusing. The change was unrelated to the feature. If at one point it was changed to add a new type/parameter in then I would understand. Thank you for considering the reversion and following through.